src/openrct2-ui/windows/Changelog.cpp

```diff
@@ -41,273 +41,306 @@ static constexpr const rct_string_id WINDOW_TITLE = STR_CHANGELOG_TITLE;
+    /**
+     * @brief Set the Changelog Window's Personality, should be called just after creation. Returns true on success
+     *
+     * @param personality
+     */
+    bool SetPersonality(int personality)
     {
-        case WV_NEW_VERSION_INFO:
-            if (!GetContext()->HasNewVersionInfo())
-            {
-                return nullptr;
-            }
-
-            _persnality = WV_NEW_VERSION_INFO;
-            window_new_version_process_info();
-            enabled_widgets = (1 << WIDX_CLOSE) | (1 << WIDX_OPEN_URL);
-            window_changelog_widgets[WIDX_OPEN_URL].type = WindowWidgetType::Button;
-            break;
-
-        case WV_CHANGELOG:
-            if (!window_changelog_read_file())
-            {
-                return nullptr;
-            }
-
-            _persnality = WV_CHANGELOG;
-            enabled_widgets = (1 << WIDX_CLOSE);
-            break;
-
-        default:
-            log_error("Invalid personality for changelog window: %d", personality);
-            return nullptr;
+        switch (personality)
+        {
+            case WV_NEW_VERSION_INFO:
+                if (!GetContext()->HasNewVersionInfo())
+                {
+                    return false;
+                }
+                _personality = WV_NEW_VERSION_INFO;
+                NewVersionProcessInfo();
+                enabled_widgets = (1 << WIDX_CLOSE) | (1 << WIDX_OPEN_URL);
+                widgets[WIDX_OPEN_URL].type = WindowWidgetType::Button;
+                return true;
+
+            case WV_CHANGELOG:
+                if (!ReadChangelogFile())
+                {
+                    return false;
+                }
+                enabled_widgets = (1 << WIDX_CLOSE);
+                _personality = WV_CHANGELOG;
+                return true;
+
+            default:
+                log_error("Invalid personality for changelog window: %d", personality);
+                return false;
+        }
     }
 
-    int32_t screenWidth = context_get_width();
-    int32_t screenHeight = context_get_height();
-
-    window = WindowCreateCentred(
-        screenWidth * 4 / 5, screenHeight * 4 / 5, &window_changelog_events, WC_CHANGELOG, WF_RESIZABLE);
-    window->widgets = window_changelog_widgets;
-    window->enabled_widgets = enabled_widgets;
-
-    WindowInitScrollWidgets(window);
-    window->min_width = MIN_WW;
-    window->min_height = MIN_WH;
-    window->max_width = MIN_WW;
-    window->max_height = MIN_WH;
-
-    return window;
-}
+    void OnOpen() override
+    {
+        widgets = s_wndChangelogWidgets;
 
-static void window_changelog_close([[maybe_unused]] rct_window* w)
-{
-    window_changelog_dispose_data();
-}
+        WindowInitScrollWidgets(this);
+        min_width = MIN_WW;
+        min_height = MIN_WH;
+        max_width = MIN_WW;
+        max_height = MIN_WH;
+    }
 
-static void window_changelog_mouseup(rct_window* w, rct_widgetindex widgetIndex)
-{
-    switch (widgetIndex)
+    void OnClose() override
```

Comment:
Sorry I overlooked this part, you can remove this entirely including the DisposeChangelogData function, when the window is closed the destructor takes care of it.

Reply:
Done